src/components/Input/Input.tsx

```diff
@@ -0,0 +1,177 @@
+  border-radius: 7px;
+  background: ${Color.BackgroundMain};
+  font-size: 17px;
+  cursor: ${(p) => (p.disabled ? "default" : "pointer")};
+
+  :disabled {
+    border: 2px solid ${Color.BorderDisabled};
+    background: ${Color.BackgroundDisabled};
+  }
+`;
+
+const IconContainer = styled(Row)<{ disabled?: boolean }>`
+  ${(p) =>
+    p.disabled &&
+    css`
+      svg * {
+        fill: ${Color.DisabledGrey};
+      }
+    `}
+`;
+
+// Pass accurate input type
+// https://www.w3.org/WAI/tutorials/forms/validation/#validating-common-input
+
+export const Input: FC<InputProps> = ({
+  className,
+  disabled = false,
+  label,
+  onInputChange,
+  name,
+  tabIndex,
+  type = "text",
+  required,
+  maxLength,
+  pattern,
+  size = "large",
+  width = "300px",
+  icon,
+  hasError = false,
+  isValid = false,
+  errorText = "Input not valid",
+  value,
+}: InputProps) => {
+  const [hasFocus, setHasFocus] = useState<boolean>(false);
+
+  // TODO - don't use disabled, similar to Button.tsx
+  // 'disabled' removes element for screen readers, so for a11y it's best to visually make them disbaled and use aria-diabled instead - // aria-disabled={disabled}
+  // https://a11y-101.com/development/aria-disabled
+
+  const sizeNumber = size === "medium" ? 50 : 56;
+
+  const setFocus = () => {
+    setHasFocus(true);
+  };
+
+  const removeFocus = () => {
+    setHasFocus(false);
+  };
+
+  return (
+    <Column className={className} position="relative" width={width}>
+      <LabelContainer disabled={disabled} hasFocus={hasFocus || value} hasIcon={icon !== "undefined"} size={sizeNumber}>
+        <IconContainer alignItems="center" disabled={disabled}>
+          {icon && <Box mr="8px">{icon}</Box>}
+          <StyledLabel disabled={disabled} hasError={hasError} hasFocus={hasFocus || value} htmlFor={name}>
+            {label}
+            {required && "*"}
+          </StyledLabel>
+        </IconContainer>
+      </LabelContainer>
+      <StyledInput
+        aria-invalid={hasError}
+        aria-required={required}
+        autoFocus={hasError}
+        disabled={disabled}
+        id={name}
+        maxLength={maxLength}
+        name={name}
+        pattern={pattern}
+        required={required}
+        size={sizeNumber}
+        tab-index={tabIndex}
+        type={type}
+        value={value}
+        onBlur={() => removeFocus}
+        onChange={(e) => onInputChange(e.target.value)}
+        onFocus={() => setFocus}
+      />
+      {hasError && (
+        <Row alignItems="center" height={50} mr={5} position="absolute" right={0}>
+          <ErrorIcon fill={Color.ErrorRed} height={25} width={25} />
+        </Row>
+      )}
+      {isValid && (
+        <Row alignItems="center" height={50} mr={5} position="absolute" right={0}>
+          <CheckIcon fill={Color.SecondaryGreen} height={25} width={25} />
+        </Row>
+      )}
+      {hasError && (
+        <>
+          <Spacer mt={1} />
+          <ErrorText>{errorText}</ErrorText>
```

Comment:
What about making this styled-component absolute positioned therefore when it appears it does not push/reposition the adjacent elements.

Reply:
thanks, I added that and also fixed the Error Text font size, since it is supposed to be 14 and I had 17.